modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloVelocityMojo.java

```diff
@@ -0,0 +1,86 @@
+package org.codehaus.modello.maven;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.codehaus.modello.plugin.velocity.VelocityGenerator;
+
+/**
+ * Creates files from the model using Velocity templates.
+ *
+ * @author <a href="mailto:brett@codehaus.org">Brett Porter</a>
+ */
+@Mojo( name = "velocity", defaultPhase = LifecyclePhase.GENERATE_SOURCES, threadSafe = true )
+public class ModelloVelocityMojo
+        extends AbstractModelloGeneratorMojo
+{
+    /**
+     * The output directory of the generated XML Schema.
+     */
+    @Parameter( defaultValue = "${project.build.directory}/generated-sources/modello", required = true )
+    private File outputDirectory;
+
+    @Parameter
+    private List<String> templates;
```

Comment:
Some description would be nice, is this relative file paths or resource path

Reply:
Can't we use File instead to make it clear which separators to use here (those are automatically resolved to the project's basedir by Maven)?

Reply:
They are not resolved by maven.  I'm not entirely sure if they can be used with an extended syntax or not.  The thing that is certain is that relative paths will be resolved against the project's basedir, but that's done internally in velocity as pointed in the javadoc.

Reply:
So velocity has several resource loaders, see http://dev.cs.ovgu.de/java/velocity/docs/developer-guide.html#Resource%20Loaders
However, I think they have to be configured explicitly, so given we don't do anything special, I suppose the default `FileResourceLoader` is configured and we do pass the base directory for it to resolve resources properly.

Reply:
I think it would be safer to rely on File here and just pass the absolute value to Velocity. I don't see the need for any other loaders here.

Reply:
I found a way.

Reply:
Note that templates can be loaded from templates, see https://github.com/apache/maven/blob/master/api/maven-api-model/src/main/mdo/model.vm#L19
Those will be resolved from the `basedir`...

Reply:
> I don't think that this line https://github.com/apache/velocity-engine/blob/e20a3604ec7f279daab32f6c1c86b4bfc18bc8d3/velocity-engine-core/src/main/java/org/apache/velocity/runtime/resource/loader/FileResourceLoader.java#L368 will modify a given absolute template path.
> See https://docs.oracle.com/javase/7/docs/api/java/io/File.html#File(java.lang.String,%20java.lang.String).

Not that one, but the 3 lines above remove the leading '/', so the path is always relative :-(